### PR TITLE
Fixing some core guideline warnings

### PR DESCRIPTION
**Pull request by adam4813**
Most are just size conversion warnings.

Removed a lot of defaultable constructors.

### CODE REVIEW

File: client/graphics/shader.hpp [hunk 1]
```diff
@@ -140,9 +140,9 @@ namespace tec {
 		* \param const std::string filename The filename from were the source comes (only for debug purposes)
 		* \return void
 		*/
-		void LoadFromString(const ShaderType type, const std::string& source, const std::string& filename);
+		void LoadFromString(const Shader::ShaderType type, const std::string& source, const std::string& filename);
 
-		GLuint program;
+		GLuint program{ 0 };
```
Comment (milesrout) [inline client/graphics/shader.hpp]: I don't particularly like this. What's wrong with `GLuint program = 0;`?

File: client/graphics/lights.hpp [hunk 2]
```diff
@@ -13,23 +13,17 @@
...
+		glm::vec3 color{ 1.f, 1.f, 1.f };
+		float ambient_intensity{ 0.5f };
+		float diffuse_intensity{ 0.2f };
+
+		BaseLight() = default;
 	};
 
 	struct DirectionalLight : public BaseLight {
-		glm::vec3 direction;
+		glm::vec3 direction{ 0.f, 0.f, -1.f };
```
Comment (milesrout) [inline client/graphics/lights.hpp]: In this case I'm okay with using brace initialisation, but I'd still prefer `glm::vec3 direction = {0.0f, 0.0f, -1.0f};`

I'm also not really a fan of leaving out the `0` there just because we can. I wrote the above as `0.1f, 0.0f, -1.0f` at first and had to look at this line three times before I realised it's just `0`.

File: client/graphics/animation.hpp [hunk 3]
```diff
@@ -8,7 +8,7 @@
 #include <glm/glm.hpp>
 #include <glm/ext.hpp>
 
-#include "components.pb.h"
+#include <components.pb.h>
```
Comment (milesrout) [inline client/graphics/animation.hpp]: Does this sort out the warnings being generated based on protobuf's code?

File: client/graphics/lights.hpp [hunk 4]
```diff
@@ -13,23 +13,17 @@
...
-		BaseLight() {
-			color = glm::vec3(1.0f, 1.0f, 1.0f);
-			ambient_intensity = 0.5f;
-			diffuse_intensity = 0.2f;
-		}
+		glm::vec3 color{ 1.f, 1.f, 1.f };
+		float ambient_intensity{ 0.5f };
+		float diffuse_intensity{ 0.2f };
+
+		BaseLight() = default;
```
Comment (milesrout) [inline client/graphics/lights.hpp]: Wow that's much better than before. 

I would prefer copy-initialization or copy-list-initialization for readability though.

File: client/graphics/material.cpp [hunk 5]
```diff
@@ -2,12 +2,11 @@
 // Licensed under the terms of the LGPLv3. See licenses/lgpl-3.0.txt
 
 #include "material.hpp"
+
 #include "texture-object.hpp"
 #include "shader.hpp"
 
 namespace tec {
-	Material::Material() : polygon_mode(GL_FILL), draw_elements_mode(GL_TRIANGLES) { }
-
```
Comment (milesrout) [inline client/graphics/material.cpp]: being able to just remove this totally is so so nice

File: client/gui/console.hpp [hunk 6]
```diff
@@ -53,9 +53,9 @@ namespace tec {
 
 	private:
 		// TODO Store a RingBuffer<tuple<color, text> instead of raw text ?
-		tec::RingBuffer< std::tuple< ImVec4, std::string >, 4096> buf;
+		std::shared_ptr<tec::RingBuffer< std::tuple< ImVec4, std::string >, 4096>> buf;
```
Comment (milesrout) [inline client/gui/console.hpp]: Does this need to be a shared ptr?

File: client/render-system.cpp [hunk 7]
```diff
@@ -217,7 +218,7 @@ namespace tec {
 
 		glBindVertexArray(this->sphere_vbo.GetVAO());
 		glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, this->sphere_vbo.GetIBO());
-		std::size_t index_count = this->sphere_vbo.GetVertexGroup(0)->index_count;
+		auto index_count{ static_cast<GLsizei>(this->sphere_vbo.GetVertexGroup(0)->index_count) };
```
Comment (milesrout) [inline client/render-system.cpp]: This is really ugly compared to what was there before. I assume it had a conversion warning or something like that?

File: client/render-system.cpp [hunk 8]
```diff
@@ -237,7 +238,7 @@ namespace tec {
 			}
 
 			glm::mat4 transform_matrix = glm::scale(glm::translate(glm::mat4(1.0), position) *
-				glm::mat4_cast(orientation), scale);
+													glm::mat4_cast(orientation), scale);
```
Comment (milesrout) [inline client/render-system.cpp]: this is a very big indent

File: client/resources/vorbis-stream.cpp [hunk 9]
```diff
@@ -1,45 +1,45 @@
...
-	#pragma warning( pop ) 
-	#pragma warning( pop )
+#pragma warning( pop ) 
+#pragma warning( pop ) 
+#pragma warning( pop ) 
+#pragma warning( pop ) 
+#pragma warning( pop )
 #else
-	#include "resources/stb_vorbis.c"
+#include "stb_vorbis.c"
```
Comment (milesrout) [inline client/resources/vorbis-stream.cpp]: I think these should be indented to match the nested if/else structure.

File: common/components/lua-script.cpp [hunk 10]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) 2013-2016 Trillek contributors. See AUTHORS.txt for details
 // Licensed under the terms of the LGPLv3. See licenses/lgpl-3.0.txt
 
-#include "components/lua-script.hpp"
+#include "lua-script.hpp"
```
Comment (milesrout) [inline common/components/lua-script.cpp]: are these changes required by a change to the build system? or were both paths valid before? because I definitely don't think that both paths should be valid...

File: client/graphics/animation.hpp (hunk 3 above)
Comment (adam4813) [reply client/graphics/animation.hpp]: Sadly on windows nope.

File: client/graphics/shader.hpp (hunk 1 above)
Comment (adam4813) [reply client/graphics/shader.hpp]: The reason being it is the now recommended way to do initialization https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#es23-prefer-the--initializer-syntax

File: client/gui/console.hpp (hunk 6 above)
Comment (adam4813) [reply client/gui/console.hpp]: The allocation was huge, so I just defaulted to shared. We can certainly look if it can be a unique

File: client/render-system.cpp (hunk 7 above)
Comment (adam4813) [reply client/render-system.cpp]: yes

File: client/render-system.cpp (hunk 8 above)
Comment (adam4813) [reply client/render-system.cpp]: What in the world...

File: common/components/lua-script.cpp (hunk 10 above)
Comment (adam4813) [reply common/components/lua-script.cpp]: The header is located relative to the source, so both paths are valid and both were valid.

File: client/graphics/shader.hpp (hunk 1 above)
Comment (milesrout) [reply client/graphics/shader.hpp]: Lol these guidelines were written by insane people.

File: client/resources/vorbis-stream.cpp (hunk 9 above)
Comment (adam4813) [reply client/resources/vorbis-stream.cpp]: This is a good springboard to adding clang-format, yes?

File: client/graphics/lights.hpp (hunk 2 above)
Comment (adam4813) [reply client/graphics/lights.hpp]: For future I will include both sides.

**Comment by milesrout** [comment]
merged manually
